### DISCUSSION TITLE
two byte variable handled as one short block

### DISCUSSION THREAD

**Discussion by em-eight** [NationalSecurityAgency/ghidra#6487] [Q&A]
Hello,

I'm using ghidra with an architecture that contains some aliasing between registers. Simply put, it has 8-bit registers that alias with 16-bit registers.

The issue is sometimes this causes computation to happen in 16-bit register with ugly concat pcode, when in reality it was just two 8-bit variables.

Example:
![image](https://github.com/NationalSecurityAgency/ghidra/assets/107326447/25cf13ec-69fd-4148-a6d6-e0d1cb757910)

Here uVar5 should have been split into two u8 variables.

Is there anything I can do to fix this?

**Comment by Wall-AF** [comment]
Have you tried to manually set `uVar5` as `byte[2]` or a struct containing 2 bytes?

**Reply by em-eight** [reply]
That does get rid of the CONCAT but still no separate variable
